third_party/amd/lib/TritonAMDGPUDialectToLLVM/ExtractSliceOpToLLVM.cpp

```diff
@@ -120,13 +125,97 @@ struct ExtractSliceOpConversion
     return success();
   }
 
+  // This handles extract_slice when the layout is ttg.slice.
+  // For this case, the offsets, strides and sizes are only 1d,
+  // and the sizePerThread is takes from the parentLayout.
+  // ShapePerCTATile already examines the parentLayout for sliceLayout.
+  LogicalResult processLayout1d(amdgpu::ExtractSliceOp op, OpAdaptor adaptor,
```

Comment:
We have `processLayout1d` and `processLayout2d`. Could we make a single generic function which can handle the `Nd` case?

Reply:
Good question. 1d vs 2d may be a bit of a misnomer. The 2D case handles 3 types of distributed layouts, whereas the "1d" case handles the slice layout. So in this regard there is a fair bit of divergence between the two. They have probably 50% code in common and 50% different. I could also rename it from "1d" to SliceLayout.

Reply:
> They have probably 50% code in common and 50% different.

It would be better to unify those 50% of common code.

> I could also rename it from "1d" to SliceLayout.

This would be better and less confusing.